flink-python/dev/dev-requirements.txt

```diff
@@ -15,7 +15,7 @@
 pip>=20.3
 setuptools>=18.0
 wheel
-apache-beam>=2.43.0,<2.49.0
+apache-beam>=2.54.0
```

Comment:
I think we should set the version of apache-beam to a fixed version or a specific range. Flink users usually don't upgrade their cluster frequently. We could not guarantee future apache-beam works for old Flink versions. (Take this PR as an example, it has done quite a few code changes instead of just upgrading the beam version)

Reply:
agreed. Fixed version might be too restrictive though. I see that other dependencies set the next major version as upper limit, perhaps that's too risky as well. I can try to install the current latest version locally to see if it works, and set that as upper limit, how's that?

Reply:
@xaniasd Sure. Could you also try the lower limit?

Reply:
I tried both 2.54 and 2.61, all tests pass locally. I'll use this range then.

Reply:
Updated @dianfu let me know if there's anything else to fix